includes/api/class-wc-rest-shipping-zones-controller.php

```diff
@@ -0,0 +1,250 @@
+<?php
+/**
+ * REST API Shipping Zones controller
+ *
+ * Handles requests to the /shipping/zones endpoint.
+ *
+ * @author   WooThemes
+ * @category API
+ * @package  WooCommerce/API
+ * @since    2.7.0
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * REST API Shipping Zones class.
+ *
+ * @package WooCommerce/API
+ * @extends WC_REST_Shipping_Zones_Controller_Base
+ */
+class WC_REST_Shipping_Zones_Controller extends WC_REST_Shipping_Zones_Controller_Base {
+
+	/**
+	 * Register the routes for Shipping Zones.
+	 */
+	public function register_routes() {
+		register_rest_route( $this->namespace, '/' . $this->rest_base, array(
+			array(
+				'methods'             => WP_REST_Server::READABLE,
+				'callback'            => array( $this, 'get_items' ),
+				'permission_callback' => array( $this, 'get_items_permissions_check' ),
+			),
+			array(
+				'methods'             => WP_REST_Server::CREATABLE,
+				'callback'            => array( $this, 'create_item' ),
+				'permission_callback' => array( $this, 'update_items_permissions_check' ),
```

Comment:
Just found another one: This should have it's own `create_item_permissions_check` (errors are different).